apps/fluent-tester/src/E2E/common/BasePage.ts

```diff
@@ -1,13 +1,39 @@
-import { Keys } from './consts';
+import { Keys, ROOT_VIEW } from './consts';
 import { TESTPAGE_BUTTONS_SCROLLVIEWER } from '../../TestComponents/Common/consts';
 import { Attribute } from './consts';
 
 const DUMMY_CHAR = '';
+// The E2ETEST_PLATFORM environment variable should be set in the beforeSession hook in the wdio.conf file for the respective platform
+const PLATFORM = process.env['E2ETEST_PLATFORM'];
+if (PLATFORM) {
+  console.log(`Using platform: ${PLATFORM}`);
+}
 export const COMPONENT_SCROLL_COORDINATES = { x: -0, y: -100 }; // These are the offsets. Y is negative because we want the touch to move up (and thus it scrolls down)
 
+let rootView: WebdriverIO.Element | null = null;
+
 /* Win32/UWP-Specific Selector. We use this to get elements on the test page */
-export function By(identifier: string) {
-  return $('~' + identifier);
+export async function By(identifier: string) {
+  if (PLATFORM === NativePlatform.Windows) {
+    // For some reason, the rootView node is never put into the element tree on the UWP tester. Remove this when fixed.
+    return await $('~' + identifier);
+  }
+  return await QueryWithChaining(identifier);
+}
+
+async function QueryWithChaining(identifier) {
+  if (rootView === null) {
+    // Most of the elements we're searching for will be children of this rootView node.
+    rootView = await $('~' + ROOT_VIEW);
+  }
+  const selector = '~' + identifier;
+  let queryResult: WebdriverIO.Element;
+  queryResult = await rootView.$(selector);
+  if (queryResult.error) {
+    // In some cases, such as opened ContextualMenu items, the element nodes are not children of the rootView node, meaning we need to start our search from the top of the tree.
```

Comment:
Can we find the parent of these ContextualMenu's? They must have some parent. I'm assuming this is specific to Callouts.

@PPatBoyd Are Callout's not apart of the same root window tree? It seems as though it has a different root completely.

@lawrencewin We should be able to find the parent. Once we do, we should try to set that as the ROO_VIEW for that specific `By()` call.

If we have to search for each callout from the root, the performance will be very bad on larger UI trees

Reply:
![image](https://user-images.githubusercontent.com/15683103/197075889-44d7b2e3-4d34-4bf3-880d-cf3aeab1f6dd.png)
ContextualMenuItems and the rest of the app share a parent in the Tester Window, but there isn't an automation ID I can use to query parent items.

Reply:
Perhaps we can use a [different selector strategy ](https://webdriver.io/docs/selectors/)to find this "FluentUI Tester" window component?

If we can, that would be the most ideal solution, and it should work for all platforms.

Reply:
@samuelfreiberg @ksiler Okay so after some experimentation with xpath, class name, and name, it looks like there isn't a single selector strategy that can work across all platforms.

With that said, it is definitely possible to find the root Fluent Tester window element if we write platform specific webdriver code. I've had success locating the window on win32 and uwp. Have not tested on iOS, mac, or android. It's janky, but it allows us to remove chained awaits and removes the conditional for ContextualMenu.